include/systems/generic_projector.h

```diff
@@ -2122,9 +2119,14 @@ void GenericProjector<FFunctor, GFunctor, FValue, ProjectionAction>::ProjectEdge
           FEInterface::dofs_on_edge(&elem, dim, base_fe_type,
                                     context.edge, edge_dofs);
 
-          this->construct_projection
-            (dof_indices, edge_dofs, var_component,
-             &edge_node, proj_fe);
+          if (FEInterface::field_type(proj_fe.get_fe_type()) == TYPE_SCALAR)
+            this->construct_projection
+              (dof_indices, edge_dofs, var_component,
+               &edge_node, static_cast<const FEGenericBase<Real> &>(proj_fe));
+          else
+            this->construct_projection
+              (dof_indices, edge_dofs, var_component,
+               &edge_node, static_cast<const FEGenericBase<RealVectorValue> &>(proj_fe));
```

Comment:
Could we use libmesh_cast here?  I can't see these casts failing but it'd be nice to double check when debugging anyway.